torch/csrc/jit/python_custom_operator.cpp

```diff
@@ -0,0 +1,48 @@
+#include <torch/csrc/jit/python_custom_operator.h>
+
+#include <torch/csrc/jit/ivalue.h>
+#include <torch/csrc/jit/operator.h>
+#include <torch/csrc/jit/pybind_utils.h>
+#include <torch/csrc/jit/stack.h>
+#include <torch/csrc/jit/type.h>
+#include <torch/csrc/utils/pybind.h>
+
+#include <ATen/Error.h>
+
+#include <cstddef>
+#include <vector>
+#include <utility>
+
+namespace torch { namespace jit {
+py::object invoke(const Operator& op, py::args args, py::kwargs kwargs) {
+  try {
+    // Create a stack full of the arguments and keyword arguments.
+    auto stack =
+        createStackForSchema(op.schema(), std::move(args), std::move(kwargs));
+
+    // Invoke the operation, which puts the return values onto the stack.
+    op.getOperation()(stack);
+
+    if (stack.empty()) {
+      return py::none();
+    }
+
+    // Return a simple value and not a single-element tuple if there is only one
+    // return value.
+    if (stack.size() == 1) {
+      return toPyObject(pop(stack));
+    }
+
+    // If there is more than one return value, pop them into a py::tuple.
+    py::tuple return_values(stack.size());
+    for (size_t ret = 1; ret <= return_values.size(); ++ret) {
+      return_values[return_values.size() - ret] = toPyObject(pop(stack));
```

Comment:
Argument and return values on the stack just appear in standard order:

```
c, d = foo(a,b)
before call: a b <- top of stack
after call: c, d <- top of stack
```
So this should just be `0..stack.size()`